Zend/zend_vm_def.h

```diff
@@ -7503,6 +7503,8 @@ ZEND_VM_HANDLER(194, ZEND_ARRAY_KEY_EXISTS, CV|TMPVAR|CONST, CV|TMPVAR|CONST)
 ZEND_VM_C_LABEL(array_key_exists_array):
 		ht = Z_ARRVAL_P(subject);
 		result = zend_array_key_exists_fast(ht, key OPLINE_CC EXECUTE_DATA_CC);
+	} else if (UNEXPECTED(Z_TYPE_P(subject) == IS_OBJECT && zend_class_implements_interface(Z_OBJCE_P(subject), zend_ce_arrayaccess))) {
```

Comment:
Can you make this the last condition, to avoid a performance regression for references?